### PR TITLE
CSV upload

### PR DESCRIPTION
**Pull request by macocha**
Still need to add:
- [x] tests
- [x] pre-run csv upload

@epaulet

### CODE REVIEW

File: rainforest-cli.go [hunk 1]
```diff
@@ -26,8 +23,16 @@ var (
...
+	// default output for printing resource tables
+	tablesOut io.Writer = os.Stdout
+
+	// Run status polling interval
+	runStatusPollInterval = time.Second * 5
+
+	// Batch size (number of rows) for tabular var upload
+	tabularBatchSize = 50
+	// Concurent connections when uploading CSV rows
+	tabularConcurency = 1
```
Comment (epaulet) [inline rainforest-cli.go]: Typo, but also is there any way to change this / do we want to allow that?

File: rainforest-cli.go [hunk 2]
```diff
@@ -126,12 +131,12 @@ func main() {
 					Usage: "use the abort option to abort any runs in the same environment or " +
 						"use the abort-all option to abort all runs in progress.",
 				},
-				cli.BoolTFlag{
+				cli.BoolFlag{
 					Name: "fg, foreground",
```
Comment (epaulet) [inline rainforest-cli.go]: Leaving this as false by default for now?

File: rainforest/tabularvars.go [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+	"strconv"
+	"time"
+)
+
+// Generator is a type representing generators which can be used as variables in RF tests.
+// They can be builtin or uploaded by customer as a tabular variable.
+type Generator struct {
+	ID           int               `json:"id,omitempty"`
+	Name         string            `json:"name,omitempty"`
+	CreationDate time.Time         `json:"created_at, omitempty"`
```
Comment (epaulet) [inline rainforest/tabularvars.go]: This field isn't used at all, is it?

File: tabularvars.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	return b
+}
+
+// rowUploadWorker is a helper worker which reads batch of rows to upload from rows chan
+// and pushes potential errors through errorsChan
+func rowUploadWorker(api tabularVariablesAPI, generator rainforest.Generator,
+	columns []string, rowsChan <-chan [][]string, errorsChan chan<- error) {
+	for rows := range rowsChan {
+		error := api.AddGeneratorRowsFromTable(generator, columns, rows)
+		errorsChan <- error
```
Comment (epaulet) [inline tabularvars.go]: The other rows will still upload. Should we stop the uploads immediately on all of the workers?

File: rainforest/tabularvars.go (hunk 3 above)
Comment (macocha) [inline rainforest/tabularvars.go]: We are not using it in our CLI, but it's included in the library, as someone might want to get to that. 😃

File: rainforest-cli.go (hunk 2 above)
Comment (macocha) [inline rainforest-cli.go]: Oh yeah, but I think we should switch it around and have `bg, background` flag.

File: rainforest-cli.go (hunk 1 above)
Comment (macocha) [inline rainforest-cli.go]: It's more for us to have it in a easy accessible place if we decide to change it later on.

File: tabularvars.go (hunk 4 above)
Comment (macocha) [inline tabularvars.go]: As decided on slack, we're leaving it as is, because they'll get killed anyways when main routine exits.

**Pull request merged**